### PR TITLE
Login feature

### PR DESCRIPTION
**Pull request by leifehetland**
I have login with Google using Oauth currently working. Will probably pursue github login as well and would also do Facebook but I don't have one :sob: 
@jcockhren, @AdamSheaffer, @BlaiseGratton, @clanderson87, @jonathanhuffaker

### CODE REVIEW

File: controllers/AuthCtrl.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                            console.log("Login Failed", error);
+                        } else {
+                            console.log("Authenticated successfully with:", authData);
+                            userRefObj = authData;
+                            userRefObj.username = authData.google.displayName;
+                            ref.child("users").child(authData.uid).set(userRefObj);
+                        };
+                    })
+                }
+            }
```
Comment (AdamSheaffer) [inline controllers/AuthCtrl.js]: - you're bringing in 4 services as dependencies in your authController function, but it doesn't look like you're using any of them. Are they necessary?

- are currentUser on line 13 and userRef on 15 of the authController being used?

- userRefObj on line 23 is being saved as a global variable. Maybe that's what you want (if not, you should use `var userRefObj`) but if so, you might trying saving on the $rootScope. You guys know more about the firebase auth API than I do, so it's totally up to you.

- just a general angular note: a new controller will be created/instantiated every time you hit a route the controller is connected to. Factories and services on the other hand are singletons--meaning that the object is only ever created once, and the same instance of it gets passed around to whoever else needs it as a dependency. When you have time, you might expose the firebase reference object (on line 14) in a separate service/factory. If it's in the controller function it'll get recreated every time a new controller is created, but you'll only ever need one instance of it.

File: factories/authFactory.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+app.factory('authFactory',
+  ['$firebaseAuth',
+  '$firebaseObject',
+
+  function (
+      $firebaseAuth,
+      $firebaseObject){
+
+        var ref = new Firebase("https://survey-creator.firebaseio.com");
+        return $firebaseAuth(ref);
```
Comment (AdamSheaffer) [inline factories/authFactory.js]: Do you need $firebaseObject as a dependency?

File: partials/login.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div ng-controller="AuthCtrl as authCtrl">
```
Comment (AdamSheaffer) [inline partials/login.html]: You already attached the authController with the $routeProvider, so you shouldn't need to use the ng-controller directive anymore.

**Comment by AdamSheaffer** [comment]
Nice job. Looks good. I added a few comments on some refactoring you guys might do. Some of them might be a little nitpicky, but are worth mentioning. Let me know in the morning if you want to look at any of them together. :+1:

**Comment by clanderson87** [comment]
Thanks for the info @AdamSheaffer.  Will Def hit you up for general angular help/ best practices tomorrow. Our angular instruction was kinda... Hasty. @leifehetland, login works on my end. Good job. Maybe cut out some of those dependencies and we'll merge tomorrow after some ng-guidance.

**Comment by clanderson87** [comment]
Clean up looks good to me. Merge? @AdamSheaffer @BlaiseGratton

**Comment by BlaiseGratton** [comment]
Looks good to me

**Pull request merged**